java/src/main/java/ai/rapids/cudf/PinnedMemoryPool.java

```diff
@@ -22,109 +22,46 @@
-import java.util.TreeSet;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 import java.util.concurrent.Future;
 
 /**
- * This provides a pool of pinned memory similar to what RMM does for device memory.
+ * This is the JNI interface to a rmm::pool_memory_resource<rmm::pinned_host_memory_resource>.
  */
 public final class PinnedMemoryPool implements AutoCloseable {
   private static final Logger log = LoggerFactory.getLogger(PinnedMemoryPool.class);
-  private static final long ALIGNMENT = ColumnView.hostPaddingSizeInBytes();
 
   // These static fields should only ever be accessed when class-synchronized.
   // Do NOT use singleton_ directly!  Use the getSingleton accessor instead.
   private static volatile PinnedMemoryPool singleton_ = null;
   private static Future<PinnedMemoryPool> initFuture = null;
-
-  private final long totalPoolSize;
-  private final long pinnedPoolBase;
-  private final SortedSet<MemorySection> freeHeap = new TreeSet<>(new SortedByAddress());
-  private int numAllocatedSections = 0;
-  private long availableBytes;
-
-  private static class SortedBySize implements Comparator<MemorySection> {
-    @Override
-    public int compare(MemorySection s0, MemorySection s1) {
-      return Long.compare(s0.size, s1.size);
-    }
-  }
-
-  private static class SortedByAddress implements Comparator<MemorySection> {
-    @Override
-    public int compare(MemorySection s0, MemorySection s1) {
-      return Long.compare(s0.baseAddress, s1.baseAddress);
-    }
-  }
-
-  private static class MemorySection {
-    private long baseAddress;
-    private long size;
-
-    MemorySection(long baseAddress, long size) {
-      this.baseAddress = baseAddress;
-      this.size = size;
-    }
-
-    boolean canCombine(MemorySection other) {
-      boolean ret = (other.baseAddress + other.size) == baseAddress ||
-          (baseAddress + size) == other.baseAddress;
-      log.trace("CAN {} COMBINE WITH {} ? {}", this, other, ret);
-      return ret;
-    }
-
-    void combineWith(MemorySection other) {
-      assert canCombine(other);
-      log.trace("COMBINING {} AND {}", this, other);
-      this.baseAddress = Math.min(baseAddress, other.baseAddress);
-      this.size = other.size + this.size;
-      log.trace("COMBINED TO {}\n", this);
-    }
-
-    MemorySection splitOff(long newSize) {
-      assert this.size > newSize;
-      MemorySection ret = new MemorySection(baseAddress, newSize);
-      this.baseAddress += newSize;
-      this.size -= newSize;
-      return ret;
-    }
-
-    @Override
-    public String toString() {
-      return "PINNED: " + size + " bytes (0x" + Long.toHexString(baseAddress)
-          + " to 0x" + Long.toHexString(baseAddress + size) + ")";
-    }
-  }
+  private long poolHandle;
+  private long poolSize;
 
   private static final class PinnedHostBufferCleaner extends MemoryBuffer.MemoryBufferCleaner {
-    private MemorySection section;
+    private long address;
```

Comment:
Should address be initialized to -1?

Reply:
I am not sure it matters. Our constructor takes the argument and sets it, and I don't see how we can create a cleaner that doesn't take the argument in the constructor, as java shouldn't emit a default constructor.

Reply:
I agree.